### ISSUE TITLE
Variables within a class method with the same name as the method does not work 

### ISSUE THREAD

**Issue by shuvalov-anton** [babel/babel#864]
They work with `let`, but they crash with `var`.

```js
class Ok {
  value() {
    let value = 5
    return value;
  }
}

let ok = new Ok();
console.log(ok.value()); // 5
```

```js
class NotOk {
  value() {
    var value = 5
    return value;
  }
}

let notOk = new NotOk();
console.log(notOk.value()); // value is not defined
```

**Comment by sebmck** [comment]
Related #861. Just made it always use the old method of inferring function ids since there seem to be too many issues associated with the new method. Released as of 4.4.5, thanks!

**Comment by shuvalov-anton** [comment]
haha! You're superfast! Thank you!
